src/pages/inventory/index.tsx

```diff
@@ -0,0 +1,265 @@
+import { createClient } from '@supabase/supabase-js';
+import AddIcon from '@mui/icons-material/Add';
+
+type InventoryItem = {
+  id: number;
+  item: string;
+  type: string;
+  quantity: number;
+  category: string;
+}
+
+
+const Inventory = () => {
+
+  const [originalData, setOriginalData] = useState<InventoryItem[]>([]);
+  const [data, setData] = useState<InventoryItem[]>([]);
+  const [type, setType] = useState('');
+  const [category, setCategory] = useState('');
+  const [status, setStatus] = useState('');
+  const [search, setSearch] = useState('');
+  const [anchorType, setAnchorType] = useState<null | HTMLElement>(null);
+  const [anchorCategory, setAnchorCategory] = useState<null | HTMLElement>(null);
+  const [anchorStatus, setAnchorStatus] = useState<null | HTMLElement>(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const itemsPerPage = 10;
+
+  const indexOfLastItem = currentPage * itemsPerPage;
+  const indexOfFirstItem = indexOfLastItem - itemsPerPage;
+  const currentItems = data.slice(indexOfFirstItem, indexOfLastItem);
+
+  const handleTypeClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorType(event.currentTarget);
+  };
+  const handleCategoryClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorCategory(event.currentTarget);
+  };
+  const handleStatusClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorStatus(event.currentTarget);
+  };
+
+  const handleTypeClose = () => {
+    setAnchorType(null);
+  }
+
+  const handleMenuTypeClick = (value: string) => {
+    setType(value)
+    handleFilter();
+    handleTypeClose();
+  }
+
+  const handleCategoryClose = () => {
+    setAnchorCategory(null);
+  }
+
+  const handleMenuCategoryClick = (value: string) => {
+    setCategory(value);
+    handleFilter();
+    handleCategoryClose();
+  }
+
+  const handleStatusClose = () => {
+    setAnchorStatus(null);
+  }
+
+  const handleMenuStatusClick = (value: string) => {
+    setStatus(value);
+    handleFilter();
+    handleStatusClose();
+  }
+
+  const clearTypeFilter = () => {
+    setType('');
+  }
+  const clearCategoryFilter = () => {
+    setCategory('');
+  }
+  const clearStatusFilter = () => {
+    setStatus('');
+  }
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearch(event.target.value)
+  }
+
+  const handlePageChange = (_event: React.ChangeEvent<unknown>, value: number) => {
+    setCurrentPage(value);
+  }
+
+  const handleFilter = () => {
+    const searchFiltered = originalData.filter((row: { item: string, type: string, category: string, quantity: number; }) => {
+
+      const matchesType = type ? row.type.toLowerCase().includes(type.toLowerCase()) : true;
+
+      const matchesCategory = category ? row.category.toLowerCase().includes(category.toLowerCase()) : true;
+
+      // const matchesStatus = status ? item.status.toLowerCase().includes(status.toLowerCase()) : true; //
```

Comment:
If we leave commented out code in, can you leave a comment with a reason.

Reply:
Added reaosn